dpkg-shlibdeps shows a warning because libczmq uses pthread
symbols but is not linking against it. Add check in
configure.ac and flags in Makefile.am to fix it.

--- a/configure.ac
+++ b/configure.ac
@@ -69,6 +69,13 @@
     fi
 fi
 
+# Checks for libraries
+AC_CHECK_LIB([pthread], [pthread_create], [
+       have_pthread=yes
+       pthread_LIBS="-pthread"],
+    [AC_MSG_ERROR([pthread is required for this program])])
+AC_SUBST(pthread_LIBS)
+
 PREVIOUS_CFLAGS="${CFLAGS}"
 PREVIOUS_LIBS="${LIBS}"
 
--- a/Makefile.am
+++ b/Makefile.am
@@ -10,6 +10,7 @@
     -I$(srcdir)/include
 
 project_libs = \
+    ${pthread_LIBS} \
     ${zmq_LIBS}
 
 SUBDIRS =
